src/firefox/index.js

```diff
@@ -49,28 +50,33 @@ export type RemotePortFinderFn =
 export function defaultRemotePortFinder(
   {
     portToTry=REMOTE_PORT,
+    retriesLeft=10,
     connectToFirefox=defaultFirefoxConnector,
-  }: RemotePortFinderParams = {}
+  }: RemotePortFinderParams = {},
+
 ): Promise<number> {
   log.debug(`Checking if remote Firefox port ${portToTry} is available`);
-
-  return connectToFirefox(portToTry)
-    .then((client) => {
-      log.debug(`Remote Firefox port ${portToTry} is in use`);
-      client.disconnect();
-      // TODO: instead of throw an error, pick a new random port until
-      // one of them is available.
-      // https://github.com/mozilla/web-ext/issues/283
-      throw new WebExtError(
-        `Cannot listen on port ${portToTry} because it's in use`);
-    })
-    .catch(onlyErrorsWithCode('ECONNREFUSED', () => {
-      // The connection was refused so this port is good to use.
-      return portToTry;
-    }));
+  function tryToFindAnOpenPort() {
+    return connectToFirefox(portToTry)
+      .then((client) => {
+        log.debug(`Remote Firefox port ${portToTry} is in use
+          (retries remaining: ${retriesLeft} )`);
```

Comment:
This will show up as a line break with lots of whitespace in the log. To avoid that you can concatenate to strings. It's not pretty but I think this will make a better log:
````javascript
log.debug(
  `Remote Firefox port ${portToTry} is in use ` +
  `(retries remaining: ${retriesLeft} )`);
````